File: easypermissions/src/main/java/com/sunplus/easypermissions/helper/BaseSupportPermissionHelper.java
package com.sunplus.easypermissions.helper;

import androidx.annotation.NonNull;
import android.util.Log;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import com.sunplus.easypermissions.RationaleDialogFragmentCompat;

/**
 * Created by w.feng on 2018/10/10
 * Email: fengweisb@gmail.com
 */
public abstract class BaseSupportPermissionHelper<T> extends PermissionHelper<T> {
  private static final String TAG = "BaseSupportPermissionHe";

  BaseSupportPermissionHelper(@NonNull T host) {
    super(host);
  }

  public abstract FragmentManager getSupportFragmentManager();

  @Override
  public void showRequestPermissionRationale(@NonNull String rationale,
                                             @NonNull String positiveButton,
                                             @NonNull String negativeButton,
                                             int theme,
                                             int requestCode,
                                             @NonNull String... perms) {
    FragmentManager fm = getSupportFragmentManager();

    Fragment fragment = fm.findFragmentByTag(RationaleDialogFragmentCompat.TAG);
    if (fragment instanceof RationaleDialogFragmentCompat) {
      Log.d(TAG, "Found existing fragment, not showing rationale.");
      return;
    }

    RationaleDialogFragmentCompat
        .newInstance(rationale, positiveButton, negativeButton, theme, requestCode, perms)
        .showAllowingStateLoss(fm, RationaleDialogFragmentCompat.TAG);
  }
}

File: easypermissions/src/main/java/com/almosting/easypermissions/helper/BaseSupportPermissionHelper.kt
package com.almosting.easypermissions.helper

import android.util.Log
import androidx.fragment.app.FragmentManager
import com.almosting.easypermissions.RationaleDialogFragmentCompat

/**
 * Created by w.feng on 2018/10/10
 * Email: fengweisb@gmail.com
 */
abstract class BaseSupportPermissionHelper<T> internal constructor(host: T) :
  PermissionHelper<T>(host) {
  abstract val supportFragmentManager: FragmentManager
  override fun showRequestPermissionRationale(
    rationale: String,
    positiveButton: String,
    negativeButton: String,
    theme: Int,
    requestCode: Int,
    perms: Array<String>
  ) {
    val fm = supportFragmentManager
    val fragment = fm.findFragmentByTag(RationaleDialogFragmentCompat.TAG)
    if (fragment is RationaleDialogFragmentCompat) {
      Log.d(TAG, "Found existing fragment, not showing rationale.")
      return
    }
    RationaleDialogFragmentCompat.newInstance(
      rationale,
      positiveButton,
      negativeButton,
      theme,
      requestCode,
      perms
    ).showAllowingStateLoss(fm, RationaleDialogFragmentCompat.TAG)
  }

  companion object {
    private const val TAG = "BaseSupportPermissionHe"
  }
}